src/project/mod.rs

```diff
@@ -393,10 +399,14 @@ impl Project {
     }
 
     /// Returns the custom location of pypi-name-mapping
-    pub fn pypi_name_mapping_source(&self) -> &'static MappingSource {
-        self.manifest
-            .pypi_name_mapping_source()
-            .expect("mapping source should be ok")
+    pub fn pypi_name_mapping_source(&self) -> MappingSource {
+        self.mapping_source
```

Comment:
DOes it even make sense to use a oncecell here? Since the mapping is already just returned from the manifest?

Reply:
we are doing some transformation in self.manifest.pypi_name_mapping_source()
from user defined channels into rattler::Channel
so my idea was to cache it here